src/terminal/adapter/SixelParser.cpp

```diff
@@ -729,39 +729,100 @@ void SixelParser::_decreaseFilledBackgroundHeight(const int decreasedHeight) noe
+#pragma warning(disable : 26481) // Don't use pointer arithmetic. Use span instead (bounds.1).
+#pragma warning(disable : 26490) // Don't use reinterpret_cast (type.1).
+
 void SixelParser::_writeToImageBuffer(int sixelValue, int repeatCount)
 {
     // On terminals that support the raster attributes command (which sets the
     // background size), the background is only drawn when the first sixel value
     // is received. So if we haven't filled it yet, we need to do so now.
     _fillImageBackground();
 
+    repeatCount = std::min(repeatCount, _imageMaxWidth - _imageCursor.x);
+    if (repeatCount <= 0)
+    {
+        return;
+    }
+
+    if (sixelValue == 0)
+    {
+        _imageCursor.x += repeatCount;
+        return;
+    }
+
+    // This allows us to unsafely cast _imageBuffer to uint16_t
+    // and benefit from compiler/STL optimizations.
+    static_assert(sizeof(IndexedPixel) == sizeof(int16_t));
+    static_assert(alignof(IndexedPixel) == alignof(int16_t));
+
     // Then we need to render the 6 vertical pixels that are represented by the
     // bits in the sixel value. Although note that each of these sixel pixels
     // may cover more than one device pixel, depending on the aspect ratio.
     const auto targetOffset = _imageCursor.y * _imageMaxWidth + _imageCursor.x;
-    auto imageBufferPtr = std::next(_imageBuffer.data(), targetOffset);
-    repeatCount = std::min(repeatCount, _imageMaxWidth - _imageCursor.x);
-    for (auto i = 0; i < 6; i++)
+    const auto foreground = std::bit_cast<int16_t>(_foregroundPixel);
+    auto imageBufferPtr = reinterpret_cast<int16_t*>(_imageBuffer.data() + targetOffset);
+
+    // A aspect ratio of 1:1 is the most common and worth optimizing.
+    if (_pixelAspectRatio == 1)
     {
-        if (sixelValue & 1)
+        do
         {
-            auto repeatAspectRatio = _pixelAspectRatio;
-            do
+            // This gets unrolled by MSVC. It's written this way to use CMOV instructions.
+            // Modern CPUs have fat caches and deep pipelines. It's better to do pointless reads
+            // from memory than causing branch misprediction, as sixelValue is highly random.
+            for (int i = 0; i < 6; i++)
             {
-                std::fill_n(imageBufferPtr, repeatCount, _foregroundPixel);
-                std::advance(imageBufferPtr, _imageMaxWidth);
-            } while (--repeatAspectRatio > 0);
-        }
-        else
+                const auto test = sixelValue & (1 << i);
+                // Possibly pointless read from memory, but...
+                const auto before = imageBufferPtr[i * _imageMaxWidth];
+                // ...it allows the compiler to turn this into a CMOV (= no branch misprediction).
+                const auto after = test ? foreground : before;
+                imageBufferPtr[i * _imageMaxWidth] = after;
+            }

```

Comment:
I added some comments, but the idea here is that `pixelValue` is highly random leading to branch mispredictions when bit-testing it. We can avoid that by always loading the old bitmap value and then CMOVing between the old and new value before storing it again. With modern CPUs and their deep pipelines this runs faster.